### PR TITLE
control/controlclient: close connection after HTTPS /key request

### PR DESCRIPTION
**Pull request by shayne**
Fixes https://github.com/tailscale/tailscale/issues/4128

Took a stab at it based on the issue description. LMK if I'm off here.

### CODE REVIEW

File: control/controlclient/direct.go [hunk 1]
```diff
@@ -1217,6 +1217,10 @@ func encode(v any, serverKey, serverNoiseKey key.MachinePublic, mkey key.Machine
 func loadServerPubKeys(ctx context.Context, httpc *http.Client, serverURL string) (*tailcfg.OverTLSPublicKeyResponse, error) {
 	keyURL := fmt.Sprintf("%v/key?v=%d", serverURL, tailcfg.CurrentCapabilityVersion)
 	req, err := http.NewRequestWithContext(ctx, "GET", keyURL, nil)
+	// close connection after request since we only
+	// use the regular HTTPS channel for initial
+	// bootstrapping /key
+	req.Close = true
```
Comment (bradfitz) [inline control/controlclient/direct.go:1223]: After err check

**Review by bradfitz — COMMENTED** [review]
Commit message style: "Fixes #NNN" 

Did you verify this is a change in behavior? We didn't already close? And this fixes it?

File: control/controlclient/direct.go [hunk 2]
```diff
@@ -1220,6 +1220,10 @@ func loadServerPubKeys(ctx context.Context, httpc *http.Client, serverURL string
 	if err != nil {
```
Comment (maisem) [inline control/controlclient/direct.go:1220]: Capitalize as you were writing a sentence

https://gist.github.com/adamveld12/c0d9f0d5f0e1fba1e551#comment-sentences

**Review by maisem — APPROVED** [review]

**Comment by bradfitz** [comment]
Looks like this was already fixed. See comment https://github.com/tailscale/tailscale/issues/4128#issuecomment-1320250404

File: control/controlclient/direct.go (hunk 2 above)
Comment (shayne) [reply control/controlclient/direct.go:1223]: Resolved just before push

**Comment by shayne** [comment]
We can close this unless we think it's worth merging the transport change.

**Comment by bradfitz** [comment]
Yeah, sorry, I think we should just close this. The existing code looks like it handles this well.

**Pull request closed without merging**